concent_api/concent_api/middleware.py

```diff
@@ -39,3 +43,33 @@ def __call__(self, request):
         response = self.get_response(request)
         response['Concent-Version'] = self._concent_version
         return response
+
+
+class HandleServerErrorMiddleware(object):
+    """
+    Used to catch all unhandled exceptions and return JSON response containing error information.
+    """
+    def __init__(self, get_response):
+        self.get_response = get_response
+        self.sid = None
+
+    def __call__(self, request):
+        self.sid = transaction.savepoint()
+        response = self.get_response(request)
+        transaction.savepoint_commit(self.sid)
+        return response
+
+    def process_exception(self, request, exception):  # pylint: disable=unused-argument
+        transaction.savepoint_rollback(self.sid)
+        logging.logger.exception(exception)
+        return self._build_json_response(exception)
+
+    def _build_json_response(self, exception):  # pylint: disable=no-self-use
+        message = {"error": str(exception)}
+        if hasattr(settings, "DEBUG_INFO_IN_ERROR_RESPONSES"):
+            debug_info = settings.DEBUG_INFO_IN_ERROR_RESPONSES
```

Comment:
Please include every custom setting and its description in `base.py`. Commented out if there is no good default for it. You can say in a comment that it's by default set to `DEBUG`.

Please consider including it also in `development.py`. It won't change the functionality (`DEBUG` is already enabled in development) but it might make the setting easier to discover for someone looking for debug options.

Reply:
done